### PR TITLE
Prevent `:podspec_paths` from being swallowed when specified in .gen_config.yml

### LINKED ISSUE #45: `podspec_paths` config option cannot be set in `.gen_config.yml`
When parsing arguments from the command line, `podspec_paths` will be initialized to an empty array if no positional arguments were provided.

https://github.com/square/cocoapods-generate/blob/2dcee8858af795d1e396125b7f0a8ce0aba6cb81/lib/cocoapods/command/gen.rb#L45-L46

This hash is then passed to `merge_configuration`, which will cause any existing array which was loaded from `.gen_config.yml` to be swallowed by the empty array during `merge!` here: https://github.com/square/cocoapods-generate/blob/2dcee8858af795d1e396125b7f0a8ce0aba6cb81/lib/cocoapods/command/gen.rb#L95

I'll have a PR up soon but let me know if I missed something

### PR DESCRIPTION
**Pull request by amorde**
Closes #45

### CODE REVIEW

**Comment by segiddins** [comment]
What'll happen when it's specified both in a config file & via the command line?

**Comment by amorde** [comment]
I believe the command line args would overwrite the config value, both with and without this PR, since merge takes the last value  by default. We could add custom merging logic if we wanted to combine them, but it's not clear to me if there's a "correct" behavior there

**Comment by segiddins** [comment]
Whatever behavior we decide on, please add a test for it?

**Comment by amorde** [comment]
I've added a test for the existing behavior - open to changing it but maybe that would be better in a follow up since this PR keeps the behavior consistent when mixing the two

File: spec/cocoapods/command/gen_spec.rb [hunk 1]
```diff
@@ -304,6 +304,27 @@
             )
           end
         end
+
+        describe 'when specifying :podspec_paths in configuration files and CLI args' do
+          let(:argv) { %w[C.podspec D.podspec] }
```
Comment (segiddins) [inline spec/cocoapods/command/gen_spec.rb]: what happens if the podspecs in ARGV don't completely match what's in the local config?

Comment (amorde) [reply spec/cocoapods/command/gen_spec.rb]: 🤦‍♂ too many `A B C D`.. will update the test to make it clearer. The answer is the CLI args win and override the local config, as mentioned in the PR discussion

Comment (segiddins) [reply spec/cocoapods/command/gen_spec.rb]: 👍🏻

**Pull request merged**